middleware/dirCreation.js

```diff
@@ -0,0 +1,32 @@
+const path = require('path');
+const fs = require('fs');
+
+module.exports = function dirCreation(req, res, next) {
+  const { username, repo } = req.params;
+
+  console.log(username, repo);
+
```

Comment:
this function creates below dir structure by checking its existence

 /uploads
    /<username>
        metadata.txt
        /<repo_name>
            <file1.js>
            <file2.html>
        /<repo_name>
            <xfile1.js>
            <xfile2.html>
    /<username>
        ...